### PR TITLE
BHBC-2052

### PR DESCRIPTION
**Pull request by AlfredRosenthal**
# Overview

## Links to Jira tickets

[BHBC-2052](https://quartech.atlassian.net/browse/BHBC-2052?atlOrigin=eyJpIjoiZDIxNDcwNmE5MDA2NDA2Yjg4YjU5Zjc2YWM0NjQ4NTQiLCJwIjoiaiJ9)

## Description of relevant changes

- Recent changes for intake means the system no longer silently fails
- Issue found with intake endpoint failing with only project metadata (no occurrence information)
- Logic added to circumvent occurrence specific steps

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

File: api/src/services/dwc-service.ts [hunk 1]
```diff
@@ -83,11 +83,16 @@ export class DarwinCoreService extends DBService {
 
       await this.create_step6_transformAndUploadMetaData(submissionId, dataPackageId);
 
-      await this.create_step7_normalizeSubmissionDWCA(submissionId);
+      const isOnlyMetadata = await this.isSubmissionMetadataOnly(submissionId);
 
-      await this.create_step8_runSpatialTransforms(submissionId);
+      // skip these if the submission only contains metadata
```
Comment (NickPhura) [inline api/src/services/dwc-service.ts]: nit: can you elaborate on this comment. Include something about how these steps all involve processing the darwin core observation data, and so can be skipped if the incoming dataset doesn't include it.

File: api/src/utils/media/dwc/dwc-archive-file.ts [hunk 2]
```diff
@@ -85,6 +85,18 @@ export class DWCArchive {
     }
   }
 
+  // checks if this DwCArchive only has meta data
```
Comment (NickPhura) [inline api/src/utils/media/dwc/dwc-archive-file.ts]: nit: mind turning this into a full jsdoc comment

Comment (AlfredRosenthal) [reply api/src/utils/media/dwc/dwc-archive-file.ts]: done

File: api/src/services/dwc-service.ts (hunk 1 above)
Comment (AlfredRosenthal) [reply api/src/services/dwc-service.ts]: added a bit more to clarify

File: api/src/utils/media/dwc/dwc-archive-file.ts [hunk 3]
```diff
@@ -85,6 +85,22 @@ export class DWCArchive {
...
 
+  /**
+   * This function checks worksheet data if the DwCArchive only contains metadata
+   *
+   * @returns {boolean} True if no worksheet data is present
+   */
+  isMetaDataOnly(): boolean {
+    return (
+      this.eml != null &&
+      this.worksheets.event == null &&
```
Comment (curtisupshall) [inline api/src/utils/media/dwc/dwc-archive-file.ts]: Should we be using triple-equals instead, to check if these are strictly equal to null? If we're just checking that these properties are truthy and not if they're strictly null, maybe you could just use `!this.worksheets.event && ...`.

Comment (AlfredRosenthal) [reply api/src/utils/media/dwc/dwc-archive-file.ts]: good point. I've refined the checks

**Review by NickPhura — APPROVED** [review]

**Pull request merged**